### PR TITLE
DC/MLX5: Avoid connecting to iface on different subnet.

### LINKED ISSUE #1607: dc_x error: dc_mlx5 transport does not support global address
Command line to reproduce:

/hpc/local/benchmarks/hpcx_install_Monday/hpcx-gcc-redhat7.2/ompi-v2.x/bin/mpirun -np 1024 -mca btl_openib_warn_default_gid_prefix 0 --debug-daemons --bind-to core --tag-output --timestamp-output --display-map -mca pml ucx -x UCX_NET_DEVICES=mlx5_0:1,mlx5_1:1,mlx5_2:1,mlx5_3:1 -mca btl_openib_if_include mlx5_0:1,mlx5_1:1,mlx5_2:1,mlx5_3:1 -mca coll_hcoll_enable 0 -x UCX_TLS=dc_x,sm -mca opal_pmix_base_async_modex 0 -mca mpi_add_procs_cutoff 100000 --map-by node /hpc/scrap/users/mtt/scratch/ucx_ompi/20170612_173917_18898_748807_clx-hercules-001/installs/cMBZ/tests/imb/imb/src/IMB-MPI1 -npmin 1024 -iter 1000 -mem 0.9

dc_mlx5.c:54   UCX  ERROR dc_mlx5 transport does not support global address
pml_ucx.c:322 Error: Failed to connect to proc: 447, Destination is unreachable

http://e2e-gw.mellanox.com:4080//hpc/scrap/users/mtt/scratch/ucx_ompi/20170612_173917_18898_748807_clx-hercules-001/html/test_stdout_vWW3BO.txt

@yosefe : Need to check why wireup phase does not prevent this from happening.

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #1607

### CODE REVIEW

File: src/uct/ib/dc/accel/dc_mlx5.c [hunk 1]
```diff
@@ -51,7 +52,13 @@ static UCS_CLASS_INIT_FUNC(uct_dc_mlx5_ep_t, uct_iface_t *tl_iface,
     }
 
     if (is_global) {
-        ucs_error("dc_mlx5 transport does not support global address");
+        char buf[64];
```
Comment (brminich) [inline src/uct/ib/dc/accel/dc_mlx5.c]: minor: you declare buf here, but ah_attr (needed in this branch only) is declared with wider scope.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1896/ for details.

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3870/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3873/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
you should be able to filter by subnet id ?

**Comment by yosefe** [comment]
on most other IB transports it can connect across subnets

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3889/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3913/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3918/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
According to IBTA you can not connect across subnet. For sure you can not with RC. So, it does not make sense

**Comment by yosefe** [comment]
@shamisp you can connect over a subnet by using IB router, or when using RoCE with IP router

**Pull request merged**